src/agent.cpp

```diff
@@ -13,18 +12,32 @@ namespace
     {
         return url.defrag().escape().fullpath();
     }
+
+    std::string trim_front(const std::string& str, const char chr)
+    {
+        auto itr = std::find_if(str.begin(), str.end(),
+                       [chr](const char c) {return c != chr;});
+        return std::string(itr, str.end());
+    }
 }
 
 namespace Rep
 {
     Agent& Agent::allow(const std::string& query)
     {
+
```

Comment:
Why a blank line here?

Reply:
I have no explanation for this blank line.